hanko: init at 0.5.2

SRv6d opened the pull request:
Adds [hanko](https://github.com/SRv6d/hanko), a rust based tool that keeps git allowed signers up to date.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SRv6d committed:
maintainers: add srv6d

Noodlez1232 commented on pkgs/by-name/ha/hanko/package.nix:
```diff
@@ -0,0 +1,33 @@
+    homepage = "https://github.com/SRv6d/hanko";
+    changelog = "https://github.com/SRv6d/hanko/blob/main/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ SRv6d ];
```
The case used is wrong. It should be `[ srv6d ]` as written in the maintainers.nix file.

SRv6d replied:
Thanks, fixed.

pluiedev commented on pkgs/by-name/ha/hanko/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+  src = fetchFromGitHub {
+    owner = "SRv6d";
+    repo = pname;
```
`repo = pname;` is an antipattern as the repo name isn't inherently related to the pname in any way
```suggestion
    repo = "hanko";
```

pluiedev commented on pkgs/by-name/ha/hanko/package.nix:
```diff
@@ -0,0 +1,33 @@
+  src = fetchFromGitHub {
+    owner = "SRv6d";
+    repo = pname;
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

pluiedev commented on pkgs/by-name/ha/hanko/package.nix:
```diff
@@ -0,0 +1,33 @@
+    installShellCompletion assets/completions/hanko.fish
+  '';
+
+  meta = with lib; {
```
Avoid with lib;
```suggestion
  meta = {
```

SRv6d replied:
I saw this commonly done in other package files, is there some sort of style guide that advocates against this?

pluiedev replied:
More detailed discussion can be found at https://github.com/NixOS/nixpkgs/issues/208242 — it's a style lint done for new packages but it hasn't been implemented for older packages yet

SRv6d replied:
My bad, didn't realize I wasn't actually using it at all, thanks for the link.

SRv6d replied:
*was needed but fixed it

pluiedev commented on pkgs/by-name/ha/hanko/package.nix:
```diff
@@ -0,0 +1,33 @@
+  '';
+
+  meta = with lib; {
+    description = "Keeps your Git allowed signers file up to date with signing keys configured on software development platforms like GitHub and GitLab";
```
Descriptions should be short, concise phrases that fit into one sentence. More description text could always be put into `longDescription` instead.

```suggestion
    description = "Keeps your Git allowed signers file up to date";
```

pluiedev commented on pkgs/by-name/ha/hanko/package.nix:
```diff
@@ -0,0 +1,33 @@
+{ lib, fetchFromGitHub, installShellFiles, rustPlatform }:
```
Please reformat with `nixfmt-rfc-style`

pluiedev reviewed (changes requested):
Welcome to Nixpkgs! Here's some suggestions:

SRv6d committed:
hanko: init at 0.5.2
https://github.com/SRv6d/hanko/blob/main/CHANGELOG.md

Apply suggestions from code review

Co-authored-by: Leah Amelia Chen <github@acc.pluie.me>

Remove `with lib`

pluiedev reviewed (approved):
LGTM, just make sure to squash the fixup commits into the main commits (we don't do squash merges in Nixpkgs)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2174
